metadata-ingestion/src/datahub/ingestion/source/hex/hex.py

```diff
@@ -129,6 +205,28 @@ def __init__(self, config: HexSourceConfig, ctx: PipelineContext):
             categories_as_tags=self.source_config.categories_as_tags,
             set_ownership_from_email=self.source_config.set_ownership_from_email,
         )
+        self.project_registry: Dict[str, Project] = {}
+        self.component_registry: Dict[str, Component] = {}
+
+        self.datahub_client: Optional[DataHubClient] = None
+        self.query_fetcher: Optional[HexQueryFetcher] = None
+        if self.source_config.include_lineage:
+            assert ctx.graph, "Graph is required for lineage ingestion"
```

Comment:
```suggestion
            graph = self.ctx.require_graph(...)
```

Reply:
addressed in commit 51c65f1a0a8fc0b2d8178c328f876623d9142098